protocol/messenger_communities.go

```diff
@@ -2351,10 +2351,71 @@ func (m *Messenger) ShareCommunity(request *requests.ShareCommunity) (*Messenger
 	return response, nil
 }
 
+func (m *Messenger) ShareCommunityV2(request *requests.ShareCommunity) (*MessengerResponse, error) {
```

Comment:
Though versioning is a good idea, I think we're able to avoid this.
We introduce `ShareCommunityV2` because I warned you about `advertiseCommunityTo` used in tests. But I believe it should be possible to update it internally and not break anything.

This is the function I'm talking about:

https://github.com/status-im/status-go/blob/384543d3a6329fe88848f14daad1e560d5c89af8/protocol/communities_messenger_helpers_test.go#L322-L347

The function has access to both sender's and receiver's messengers. So we could just take required parts from the old functionality and force it's usage here. 

1. Take the community description from sender's messenger:
https://github.com/status-im/status-go/blob/384543d3a6329fe88848f14daad1e560d5c89af8/protocol/messenger.go#L2306-L2318

2. Push the community description to receiver's messenger:
https://github.com/status-im/status-go/blob/e32c5546e1e5d13f597d2c4d56c92cf831d35ea8/protocol/messenger_handler.go#L2298-L2306

wdyt?

Reply:
Here comes a more general question: _should we deprecate this message type and usage of `Messenger.CommunityID` in this PR?_

We could either do it here and not introduce a new `ShareCommunityV2`, or I could pick it up after this is merged with new endpoint and updated all other status-go places to deprecate the old functionality.

I'm ok with both ways, but please open an tracking issue if we decide to go with second option 🙂

Reply:
Kindly take a look at the new changes, If you thought they are okay then I'll remove the `protobuf.ChatMessage_COMMUNITY` type.

Reply:
Will continue investigating how to do it with any permissions tomorrow.

Reply:
> For example: This doesn't work (Non admin member does invitre)

What exactly doesn't work in this test?

Just a note, this could be related: https://github.com/status-im/status-go/issues/4413

Reply:
Cannot wrap community because sender is not admin.